src/components/constraints/ConstraintListItem.svelte

```diff
@@ -7,40 +7,155 @@
   import { Status } from '../../enums/status';
-  import type { ConstraintMetadata, ConstraintPlanSpec, ConstraintResponse } from '../../types/constraint';
+  import type {
+    ConstraintDefinition,
+    ConstraintMetadata,
+    ConstraintPlanSpecification,
+    ConstraintResponse,
+  } from '../../types/constraint';
+  import type { Argument, FormParameter } from '../../types/parameter';
+  import type { ValueSchema } from '../../types/schema';
   import { getTarget } from '../../utilities/generic';
   import { permissionHandler } from '../../utilities/permissionHandler';
   import { pluralize } from '../../utilities/text';
   import { tooltip } from '../../utilities/tooltip';
   import Collapse from '../Collapse.svelte';
   import ContextMenuItem from '../context-menu/ContextMenuItem.svelte';
+  import Parameters from '../parameters/Parameters.svelte';
   import StatusBadge from '../ui/StatusBadge.svelte';
   import ConstraintViolationButton from './ConstraintViolationButton.svelte';
 
   export let constraint: ConstraintMetadata;
-  export let constraintPlanSpec: ConstraintPlanSpec;
+  export let constraintPlanSpec: ConstraintPlanSpecification;
   export let constraintResponse: ConstraintResponse;
+  export let deletePermissionError: string = 'You do not have permission to delete constraints for this plan.';
+  export let editPermissionError: string = 'You do not have permission to edit constraints for this plan.';
   export let modelId: number | undefined;
-  export let hasReadPermission: boolean = false;
+  export let hasDeletePermission: boolean = false;
   export let hasEditPermission: boolean = false;
-  export let readOnly: boolean = false;
+  export let hasReadPermission: boolean = false;
+  export let readPermissionError: string = 'You do not have permission to view this constraint.';
   export let totalViolationCount: number = 0;
   export let visible: boolean = true;
 
   const dispatch = createEventDispatcher<{
-    toggleVisibility: { id: number; visible: boolean };
-    updateConstraintPlanSpec: ConstraintPlanSpec;
+    deleteConstraintInvocation: ConstraintPlanSpecification;
+    duplicateConstraintInvocation: ConstraintPlanSpecification;
+    toggleVisibility: { constraintId: number; invocationId: number; visible: boolean };
+    updateConstraintPlanSpec: ConstraintPlanSpecification;
+    updateConstraintPlanSpecOrder: ConstraintPlanSpecification;
   }>();
 
+  let formParameters: FormParameter[] = [];
+  let order: number;
+  let orderInput: HTMLInputElement;
   let revisions: number[] = [];
+  let upButtonHidden: boolean = false;
+  let version: Pick<ConstraintDefinition, 'type' | 'revision' | 'parameter_schema'> | undefined = undefined;
 
   $: revisions = constraint.versions.map(({ revision }) => revision);
   $: violationCount = constraintResponse?.results?.violations?.length;
   $: success = constraintResponse?.success;
+  $: {
+    order = constraintPlanSpec.order;
+    upButtonHidden = order <= 0;
```

Comment:
Trying to remember what happens with goals but should a user be allowed to modify constraint order when there are no other constraints in the list? And maybe it is too weird to restrict order and have to automatically change it when you add/remove things.. but just wanted to confirm that this is alright? 
<img width="593" alt="image" src="https://github.com/user-attachments/assets/f3ac9c54-bb17-4eef-94eb-a7d3726460e7" />

Reply:
Looks like in the model associations constraints tab the ordering is more strictly enforced so would be good to be consistent if possible

Reply:
Woah, can you give repro for this? This looks like a DB bug

Reply:
Ah, I see. And taking a deeper look, this looks 2-fold:
1) the db function isn't checking that the max order isn't the max number of constraints on the spec
2) The UI should hide the "down" option on the final constraint, same as how it hides the "up" option on the first constraint

Reply:
@AaronPlave I've pushed a fix to resolve (1) to the backend branch. You should now get a red error notif when you hit down on the final constraint (or goal).

Reply:
I also pushed a fix that doesn't show the down button when it's last

Reply:
Looks like the logic's reversed on the Scheduling side. Constraints look correct though.

<img width="347" alt="image" src="https://github.com/user-attachments/assets/15d3cff4-e6bd-42e0-ae2e-83ff1462072d" />

<img width="347" alt="image" src="https://github.com/user-attachments/assets/40199e3b-f5ff-4227-b2c0-8b9aa5158ca8" />

<img width="347" alt="image" src="https://github.com/user-attachments/assets/e57ae2f8-b759-4ef9-bf98-b4abc03db280" />

Reply:
Unresolving this conversation as the scheduling "show/hide" logic is still inversed.

Reply:
🤦🏻 fixed!

Reply:
🤦🏻 fixed